tools/dependency/validate.py

```diff
@@ -91,12 +100,16 @@ def QueryExternalDeps(self, *targets):
     deps = subprocess.check_output(['bazel', 'query', deps_query],
                                    stderr=subprocess.PIPE).decode().splitlines()
     ext_deps = set()
+    implied_untracked_deps = set()
     for d in deps:
       match = BAZEL_QUERY_EXTERNAL_DEP_RE.match(d)
       if match:
         ext_dep = match.group(1)
-        if ext_dep not in IGNORE_DEPS:
-          ext_deps.add(ext_dep)
+        if ext_dep in self._ignore_deps:
+          continue
+        if ext_dep in self._implied_untracked_deps_revmap:
+          ext_dep = self._implied_untracked_deps_revmap[ext_dep]
```

Comment:
for some reason this was hard for me to gather. could you add a small comment 
"if the dependency is untracked, add the source dependency that loaded it transitively"